verilog/formatting/formatter_test.cc

```diff
@@ -2023,13 +2026,32 @@ static const std::initializer_list<FormatterTestCase> kFormatterTestCases = {
      "logic b[K:1][W:1];\n"},
     // spaces in bit slicing
     {
+        // preserve 0 spaces
         "always_ff @(posedge clk) begin "
-        "dummy  <=\tfoo  [  7  :  2  ] ; "
+        "dummy  <=\tfoo  [  7:2  ] ; "
         "end",
         "always_ff @(posedge clk) begin\n"
         "  dummy <= foo[7:2];\n"
         "end\n",
     },
+    {
+        // preserve 1 space
+        "always_ff @(posedge clk) begin "
+        "dummy  <=\tfoo  [  7 : 2  ] ; "
+        "end",
+        "always_ff @(posedge clk) begin\n"
+        "  dummy <= foo[7 : 2];\n"
+        "end\n",
+    },
+    {
+        // limit multiple spaces to 1
+        "always_ff @(posedge clk) begin "
+        "dummy  <=\tfoo  [  7  :  2  ] ; "
+        "end",
+        "always_ff @(posedge clk) begin\n"
+        "  dummy <= foo[7 : 2];\n"
+        "end\n",
+    },
```

Comment:
Suggested test cases to demonstrate symmetrization (using left spacing), since these are harder to cover in the unit-tests:

* `7: 2` -> `7:2`
* `7 :2` -> `7 : 2`
* `7   :2` -> `7 : 2` (more than one space between `7` and `:` (github markdown rendering doesn't show it)

Reply:
Added unit tests to cover 0/1/many (2) spaces on both sides.